### PR TITLE
test(clusterconfig): add unit tests for cluster proxy gatherer

### PR DESCRIPTION
**Pull request by ncaak**
This PR implements a new unit tests for Cluster's Proxy gatherer, as well as file renaming to follow last naming convention.

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

- N/A

## Documentation

- N/A

## Unit Tests

- `pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Changelog

Yes

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-10110

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+					},
+				},
+			},
+			errorCount: 0,
+		},
+		{
+			name: "Proxy spec HTTPPROXY returns obfuscated string instead real value",
+			proxy: &v1.Proxy{
+				ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+				Spec:       v1.ProxySpec{HTTPProxy: "0.0.0.0:8443"},
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go:64]: I guess this was supposed to be `HTTPSProxy` or ? :) You can also add `NoProxy` and `ReadinessEndpoints` then...

Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go:64]: Well, the code that anonymizes those values is the same for HTTPProxy, HTTPSProxy, NoProxy and ReadinessEndpoints: [file](https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/clusterconfig/proxies.go#L58)

I wanted to do a check of it working in any value, but I can test every value. I'll update it today.

**Comment by tremes** [comment]
New unit test.
/lgtm

File: pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go (hunk 1 above)
Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_cluster_proxy_test.go:64]: Oh I didn't get it, but yes, it's confusing. I'll put a better message on the unit test to clarify that.

**Pull request merged**